lib/src/folding/parsers/highlight.dart

```diff
@@ -66,6 +66,19 @@ class HighlightFoldableBlockParser extends AbstractFoldableBlockParser {
   void _processComment(Node node) {
     final newlineCount = node.getNewlineCount();
 
+    if (_foundNonWhitespace) {
+      return;
+    }
+
+    if (node.children?.any(
+          (subnode) => BracketsStartEndNamedSectionParser.isNamedSectionTag(
```

Comment:
There may be other types of hidden comments, so we should not rely on `isNamedSectionTag()` here.

In `Code` constructor, we have `serviceComments` as the source of truth for which comments are hidden, looks like we need to pass it to `HighlightFoldableBlockParser.parse()`.

Next is the task to match nodes with the comments to see if a node's comment is a service comment.

Reply:
fixed